pkg/controller/direct/bigtable/bigtableappprofile_controller.go

```diff
@@ -0,0 +1,284 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bigtable
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	krm "github.com/GoogleCloudPlatform/k8s-config-connector/apis/bigtable/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/config"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/directbase"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/registry"
+
+	gcp "cloud.google.com/go/bigtable"
+
+	bigtablepb "cloud.google.com/go/bigtable/admin/apiv2/adminpb"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func init() {
+	registry.RegisterModel(krm.BigtableAppProfileGVK, NewBigtableAppProfileModel)
+}
+
+func NewBigtableAppProfileModel(ctx context.Context, config *config.ControllerConfig) (directbase.Model, error) {
+	return &modelBigtableAppProfile{config: *config}, nil
+}
+
+var _ directbase.Model = &modelBigtableAppProfile{}
+
+type modelBigtableAppProfile struct {
+	config config.ControllerConfig
+}
+
+func (m *modelBigtableAppProfile) client(ctx context.Context, parentProject string) (*gcp.InstanceAdminClient, error) {
+	gcpClient, err := gcp.NewInstanceAdminClient(ctx, parentProject)
```

Comment:
```
error creating instance admin client: building BigtableAppProfile client: dialing: google: could not find default credentials.
```

Could you try get the `RESTClientOptions` from the `ControllerConfig`?

e.g.
https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/pkg/controller/direct/bigquerydatatransfer/bigquerydatatransferconfig_controller.go#L59-L70